### PR TITLE
Update Seq2Seq QA example script to use SQuAD metric.

### PR DESCRIPTION
**Pull request by karthikrangasai**
# What does this PR do?

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. Discussed in [this thread](https://github.com/huggingface/transformers/pull/13432#discussion_r735561720)
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger @NielsRogge

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
This is a good switch to the SQUAD metric! I realize however that the processing is not completely adapted to what we do: the answer is generated from the first feature created by each example, so there is really no point keeping all the features.

We either need to figure out a score for a generated answer, so that we can pick the best answer corresponding to all the features generated by one example, or we can just simplify the preprocessing and only keep one feature per example.

**Comment by karthikrangasai** [comment]
In the extractive method, the logits are used to score the outputs but for generative there isn't any value we can define to rank it because the sentence is already generated. So I suppose we can just use one feature per example.

**Comment by sgugger** [comment]
It looks like GitHub does not like your rebase and is showing a diff with ~200 files touched. Could you close this PR and create a fresh one?

**Comment by karthikrangasai** [comment]
Sorry for the late reply, I didn't notice it while pushing the code.
I have created the new PR here #14335 .
I will close this one now.
